sunscreen_math/src/metal_impl/ristrettovec.rs

```diff
@@ -45,32 +45,51 @@ impl GpuRistrettoPointVec {
 
         let mut field_vec = Self { data, len };
 
-        let data_slice = unsafe { field_vec.buffer_slice_mut() };
+        // Turn the slice into a ptr into a usize so we can share it across threads.
+        // Each cell in the slice is going to be written to exactly 1 time, so a
+        // data race doesn't actually occur.
+        let data_ptr = unsafe { field_vec.buffer_slice_mut().as_ptr() } as usize;
 
-        for (i, p) in x.iter().enumerate() {
-            let x = p.0.X.to_u29();
-            let y = p.0.Y.to_u29();
-            let z = p.0.Z.to_u29();
-            let t = p.0.T.to_u29();
+        x.par_iter()
+            .map(|p| {
+                (
+                    p.0.X.to_u29(),
+                    p.0.Y.to_u29(),
+                    p.0.Z.to_u29(),
+                    p.0.T.to_u29(),
+                )
+            })
+            .enumerate()
+            .for_each(|(i, (x, y, z, t))| unsafe {
+                // Each cell in the
+                let data_ptr = data_ptr as *mut MaybeUninit<u32>;
 
-            let u29_len = x.len();
+                let u29_len = x.len();
 
-            for (j, w) in x.iter().enumerate() {
-                data_slice[(j + 0 * u29_len) * len + i].write(*w);
-            }
+                for (j, w) in x.iter().enumerate() {
+                    let elem = data_ptr.add((j + 0 * u29_len) * len + i);
 
-            for (j, w) in y.iter().enumerate() {
-                data_slice[(j + 1 * u29_len) * len + i].write(*w);
-            }
+                    (*elem).write(*w);
```

Comment:
Since we're already using pointers, the `MaybeUninit` interface makes less sense here. Writing to a pointer doesn't incur UB because you'll never attempt to drop the object already at that location.

Reply:
Let's not mess with that.